tests/unit_tests/seraphis_cache.cpp

```diff
@@ -0,0 +1,74 @@
+#include "seraphis_cache.h"
+
+#include <boost/range/adaptor/reversed.hpp>
+#include <gtest/gtest.h>
+
+bool HalfPruning::should_prune(const std::deque<uint64_t> &window, uint64_t chain_top)
```

Comment:
will not stay as a virtual, would probably be incorporated into the Cache, just wanted to open options

Reply:
I'd apply the YAGNI principle here. There is no current or predicted need for multiple pruning strategies.